core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +156,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      Option(server.getThreadPool).collect { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
Just checking, should I use `isInstanceOf` instead of `case`, which will both avoid creating a partial function as well as it's most readable to both Scala and Java dev? Also it's fewer lines of code and no need to write the quite ugly `case _ => // Do nothing`?

```scala
    val threadPool = server.getThreadPool
    if (threadPool != null && threadPool.isInstanceOf[LifeCycle]) {
      threadPool.asInstanceOf[LifeCycle].stop
    }
```

Besides, if we have to check `null` like you suggested, why not use `Option`, which is one of the most common construct Scala?